api4/system.go

```diff
@@ -451,6 +451,10 @@ func testS3(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
+	cfg := c.App.GetConfig()
+	if !*cfg.ServiceSettings.EnableLinkPreviews {
+		return
```

Comment:
I think we should be returning an empty location.
Also, please add test to cover this case.  Thanks!